include/interface/cmdTlmServer.h

```diff
@@ -0,0 +1,96 @@
+#ifndef CMDTLMSERVER_H
+#define CMDTLMSERVER_H
+
+#include <stdbool.h>
+#include <pthread.h>
+#include <glib.h>
+#include <mqueue.h>
+
+#include "circular_buffer.h"
+
+typedef struct _serverInfo {
+    // Pointer of the server name
+    char *pName;
+    // Timeout for blocking call in millisecond. If -1, means never timeout.
+    int timeout;
+    // Size of the telemetry message in byte. If there are multiple telemetry
+    // structures, use the biggest size.
+    unsigned int sizeMsgTlm;
+    // Socket to listen to the connection request
+    int socketListen;
+    // Socket to connect to the TCP/IP client
+    int socketConnect;
+    // Thread to run the server
+    pthread_t thread;
+    // Is started to listen to the command and write the telemetry or not.
+    // This value is set to be true when the thread is ready, false when the
+    // software is ready to close the server.
+    bool isStart;
+    // Server status with the enum 'ServerStatus'
+    int serverStatus;
+    // Pointer of the name of command status queue
+    char *pQueueNameCmdStatus;
+    // Message queue of the command status
+    mqd_t msgQueueCmdStatus;
+    // Pointer of the command status used in message queue
+    gpointer *pMsgCmdStatus;
+    // Pointer of the name of telemetry queue
+    char *pQueueNameTlm;
+    // Message queue of the telemetry
+    mqd_t msgQueueTlm;
+    // Pointer of the telemetry message in message queue
+    gpointer *pMsgTlm;
+    // Is the commander or not
+    bool isCommander;
+    // Command buffer to write the new command
+    cbuf_handle_t cmdMsgBuffer;
+} serverInfo_t;
+
+typedef enum {
+    // TCP/IP server is disconnected with the TCP/IP client
+    ServerStatus_Disconnected = 1,
+    // TCP/IP server is connected with the TCP/IP client
+    ServerStatus_Connected = 2,
+    // TCP/IP server exits
+    ServerStatus_Exit = 3,
+} ServerStatus;
+
+// Initialize the server.
+// The user needs to provide the following inputs:
+// - pName: Pointer of the server name
+// - timeout: Timeout for blocking call in millisecond. If the value is <=0,
+//            it will be reset to be -1 internally, which means never timeout.
+// - sizeMsgTlm: Size of the telemetry message in byte.
+// - port: Port to listen to the connection request.
+// - maxNumQueueTlm: Maximum number of telemetry messages that can be queued for
+//                   send at once.
+// - cmdMsgBuffer: Command buffer to write the new commands.
+// The server imformation will be put in 'pServerInfo'.
+// Return 0 if success, otherwise, return -1.
+int cmdTlmServer_init(serverInfo_t *pServerInfo, char *pName, int timeout,
```

Comment:
For these functions that only return 0 or -1 please consider returning a boolean instead, for consistency with many other functions in this package. (And please, please use the same meaning for true/false for all of them).

Unfortunately if you are using true = OK everywhere in this package then I think it flips the sign of the return value for these functions. I'd argue it's well worth doing, but it will require some work to accommodate.

Reply:
I am not sure this is a good idea or not. For the return value of functions in `socket.h`, `mqueue.h`, `glib.h`, and `pthread.h` (I used these in the server related code), the data type is `int` instead of `bool`. For me, this is very weird to change the data type to be `bool` instead (means the style will not be coherent anymore). Of course I understand this suggestion is to simply the return value, and it is normal to use the `true` or `false`. But I felt this is unnecessary and is not the convention (or coherent) in C language.

In addition, even though we want to change to use the `bool`, it should be `false` if the function works (in `int`, the value is `0`), and `true` if the function fails (in `int`, the value is not `0`). Therefore, for the functions in `cmdTlmServer.h`, we could return `false` if succeeds and `true` if there is some problem happens (if you really want to use the `bool`). This is more consistent with the C language style (return `0` if everything works correctly).

For the functions that return `bool` in this package, there are specific reasons. For example, in `utility.h`, you could see I name functions such as `isAbsPath()` or `isLimitSwitchOn()`. They begin with `isXXX` to do the self-explanation. For the `circular_buffer.h`, the `circular_buf_put()` and `circular_buf_get()`, they will return `false` in normal. Only when some special cases happen, the `true` will be returned (which were suggested by you in the previous PR to modify the original `int` return type).

Thanks!